sycl/doc/extensions/Reduction/Reduction_status.md

```diff
@@ -0,0 +1,189 @@
+        &GroupSum[0]);
+
+    // Store the partial sum for the work-group to the final/global reduction variable 
+    sycl::detail::workGroupBarrier();
+    if (LID == 0) {
+      Reducer.MValue = GroupSum[0];
+      Reducer.template atomic_combine(Reduction::getOutPointer(Out));
+    }
+  });
+```
+Variants (B) and (C) use the same approach. The only difference is how the partial sums accumulated per work-items are combined into a partial sum for the work-group. Instead of fast atomic to a local scalar either the `fast reduce` or `tree-reduce-loop` used there. 
+
+---
+
+TODO #4 (Performance): The `reductionLoop()` has some order in which it choses indexes from the global index space. Currently it has huge stride to help vectorizer and get more vector insturction for the device code, which though may cause competition among devices for the memory due to pretty bad memory locality. On two-socket server CPUs using smaller stride to prioritize better memory locality gives additional perf improvement. 
+
+---
+TODO #5 (Performance): Some devices may provide unique-thread-id where the number of worker threads running simultaneously is limited. Such feature opens way for more efficient implementations (up to 2x faster, especially on many stacks/tiles devices). See this extension for reference: https://github.com/intel/llvm/pull/4747
+
+---
+---
+
+# Missing `reduction` functional features to be implemented soon:
+
+### 1) Support `sycl::queue::parallel_for()` shortcuts accepting reduction variables.
+Currently, only `sycl::queue::parallel_for()` accepting `nd_range` and 1 reduction variable is supported.
+The rest of this work is temporarily blocked by XPTI instrumentation that need the last argument to be hidden from user and having a default value, which is not possible if `parallel_for` starts accepting parameter packs like this one:
+```c++
+ // Parameter pack acts as-if: Reductions&&... reductions, const KernelType &kernelFunc
+ template <typename KernelName, int Dims, typename... Rest>
+ event parallel_for(range<Dims> numWorkItems, Rest&&... rest);
+```
+
+The problem is known, the fix in SYCL headers is implemented: https://github.com/intel/llvm/pull/4352 and is waiting for some re-work in XPTI component that must be done before the fix merge.
+
+---
+### 2) Support `parallel_for` accepting `range` and having `item` as the parameter of the kernel function. 
+Currently only kernels accepting `id` are supported.
+
+---
+### 3) Support `parallel_for` accepting `range` and 2 or more reduction variables.
+Currently `parallel_for()` accepting `range` may handle only 1 reduction variable. It does not support 2 or more. 
+
+The temporary work-around for that is to use some container multiple reduction variables, i.e. std::pair, std::struct or a custom struct/class containing 2 or more reduction variables, and also define a custom operator that would be passed to `reduction` constructor.
```

Comment:
What is `std::struct`?  I'm not familiar with it, and haven't been able to find any information online.

Reply:
Misprint. I was thinking about std::tuple and struct (that is written few words later), which resulted into misprint "std::struct".
Fixed in: 55c3d7537ae5a3ac89729ee4fb88cad41873ab5d